From f47c2abf0b857cdcefd6d78731aef980579c57d3 Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Thu, 28 Jan 2016 17:59:36 +0800
Subject: [PATCH 167/538] mmc: sdhci-xenon: add quirk 2
 SDHCI_QUIRK2_TIMING_HS200_HS400

Add quirk 2 SDHCI_QUIRK2_TIMING_HS200_HS400 for A3700.
This patch solved the issue for eMMC5.0 flash that
sometimes does not work correctly.

Change-Id: I3113f8ccc40e33edd294c9c8721ca75de7fe03d1
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27682
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: Nadav Haklai <nadavh@marvell.com>
---
 drivers/mmc/host/sdhci-xenon.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mmc/host/sdhci-xenon.c b/drivers/mmc/host/sdhci-xenon.c
index ce570ba..266f943 100644
--- a/drivers/mmc/host/sdhci-xenon.c
+++ b/drivers/mmc/host/sdhci-xenon.c
@@ -992,6 +992,7 @@ static struct sdhci_pltfm_data sdhci_xenon_pdata = {
 	.quirks = SDHCI_QUIRK_NO_ENDATTR_IN_NOPDESC |
 		  SDHCI_QUIRK_MULTIBLOCK_READ_ACMD12 |
 		  SDHCI_QUIRK_NO_SIMULT_VDD_AND_POWER,
+	.quirks2 = SDHCI_QUIRK2_TIMING_HS200_HS400,
 	/* Add SOC specific quirks in the above .quirks, .quirks2
 	 * fields.
 	 */
-- 
1.9.1

